presto-main/src/main/java/com/facebook/presto/sql/gen/PageFunctionCompiler.java

```diff
@@ -296,41 +319,139 @@ private static MethodDefinition generatePageProjectMethod(ClassDefinition classD
         Variable thisVariable = method.getThis();
         BytecodeBlock body = method.getBody();
 
-        Variable from = scope.declareVariable("from", body, selectedPositions.invoke("getOffset", int.class));
-        Variable to = scope.declareVariable("to", body, add(from, selectedPositions.invoke("size", int.class)));
-        Variable positions = scope.declareVariable(int[].class, "positions");
-        Variable index = scope.declareVariable(int.class, "index");
-
         // reset block builder before using since a previous run may have thrown leaving data in the block builder
         body.append(thisVariable.setField(
                 blockBuilder,
                 thisVariable.getField(blockBuilder).invoke("newBlockBuilderLike", BlockBuilder.class, newInstance(BlockBuilderStatus.class))));
 
+        Variable pageProjectionOutput = scope.declareVariable(PageProjectionOutput.class, "pageProjectionOutput");
+        body.append(pageProjectionOutput.set(newInstance(pageProjectionOutputClass, thisVariable.getField(blockBuilder), session, yieldSignal, page, selectedPositions)))
+                .append(pageProjectionOutput.ret());
```

Comment:
These 3 lines can be simplified to:

`body.append(newInstance(pageProjectionOutputClass, thisVariable.getField(blockBuilder), session, yieldSignal, page, selectedPositions).ret)`